components/baseline/setup.ftl

```diff
@@ -271,45 +211,19 @@
           [#break]
           [#case "ssh"]
 
-            [#local localKeyPairId = subResources[LOCAL_SSH_PRIVATE_KEY_RESOURCE_TYPE].Id]
-            [#local localKeyPairPublicKey = subResources[LOCAL_SSH_PRIVATE_KEY_RESOURCE_TYPE].PublicKey]
-            [#local localKeyPairPrivateKey = subResources[LOCAL_SSH_PRIVATE_KEY_RESOURCE_TYPE].PrivateKey]
             [#local vmKeyPairId = subResources[AZURE_SSH_PRIVATE_KEY_RESOURCE_TYPE].Id]
             [#local vmKeyPairName = subResources[AZURE_SSH_PRIVATE_KEY_RESOURCE_TYPE].Name]
             [#local vmKeyVaultName = keyvaultName]
 
             [#if deploymentSubsetRequired("epilogue")]
-
+              
               [#-- Generate & Import SSH credentials into keyvault --]
+
               [@addToDefaultBashScriptOutput
                 content=[
-                  "function az_manage_ssh_credentials() {"
-                  "  info \"Checking SSH credentials ...\"",
-                  "  #",
-                  "  # Create SSH credential for the segment",
-                  "  mkdir -p \"$\{SEGMENT_OPERATIONS_DIR}\"",
-                  "  az_create_pki_credentials \"$\{SEGMENT_OPERATIONS_DIR}\" " +
-                      "\"" + regionId + "\" " +
-                      "\"" + accountObject.Id + "\" " +
-                      " ssh || return $?",
-                  "  #",
-                  "  # Update the credential if required",
-                  "  if ! az_check_key_credentials" + " " +
-                      "\"" + vmKeyVaultName + "\" " +
-                      "\"" + vmKeyPairName + "\"; then",
-                  "    pem_file=\"$\{SEGMENT_OPERATIONS_DIR}/" + localKeyPairPublicKey + "\"",
-                  "    az_update_key_credentials" + " " +
-                      "\"" + vmKeyVaultName + "\" " +
-                      "\"" + vmKeyPairName + "\" " +
-                      "\"$\{pem_file}\" || return $?",
-                  "   [[ -f \"$\{SEGMENT_OPERATIONS_DIR}/" + localKeyPairPrivateKey + ".plaintext\" ]] && ",
-                  "      { encrypt_file" + " " +
-                          "\"" + regionId + "\"" + " " +
-                          "\"" + cmkKeyId + "\"" + " " +
-                          "\"$\{SEGMENT_OPERATIONS_DIR}/" + localKeyPairPrivateKey + ".plaintext\"" + " " +
-                          "\"$\{SEGMENT_OPERATIONS_DIR}/" + localKeyPairPrivateKey + "\" || return $?; }",
-                  "  fi",
-                  "  #"
+                  "function az_generate_ssh_private_key() {"
+                    "az keyvault key create --kty RSA --size 2048 --vault-name \"" + vmKeyVaultName + "\" --name \"" + vmKeyPairName + "\"",
```

Comment:
Does this save a local private key pair? Or is that only available from in the key vault itself?

Reply:
@roleyfoley [ this latest commit ](https://github.com/codeontap/gen3-azure/pull/69/commits/fe1090a872c3fe4b8ca08f718256fe206241804b) should implement the SSH key as we discussed:

CMK - generated and stored as a Key within KeyVault
SSH - generated locally, and both public and private keys uploaded as a Secret into KeyVault.

Reply:
looks good. I think that covers off what we wanted.